### PR TITLE
Fix Issue 23553 - opCmp spec is incomplete

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

File: spec/operatoroverloading.dd [hunk 1]
```diff
@@ -487,7 +487,38 @@ $(H3 $(LNAME2 compare, Overloading $(D <), $(D <)$(D =), $(D >), and $(D >)$(D =
...
+{
+    S s;
+    const S cs;
+    B b;
+    C c;
+    assert(s > s);      // s.opCmp(s) > 0
+    assert(!(s < b));   // s.opCmp(b) > 0  - S.opCmp(ref B) is exact match
+    assert(!(b < s));   // s.opCmp(b) < 0  - S.opCmp(ref B) is exact match
+    assert(b < cs);     // b.opCmp(s) < 0  - B.opCmp(ref const S) is  exact match
+    static assert(!__traits(compiles, b < c)); // both C.opCmp and B.opcmp match exactly
```
Comment (RazvanN7) [inline spec/operatoroverloading.dd:519]: Note that, due to a bug, this assert will fail.

**Comment by RazvanN7** [comment]
cc @dkorpel @thewilsonator

File: spec/operatoroverloading.dd (hunk 1 above)
Comment (dkorpel) [reply spec/operatoroverloading.dd:519]: Is it in bugzilla?

Comment (RazvanN7) [reply spec/operatoroverloading.dd:519]: https://issues.dlang.org/show_bug.cgi?id=21453

**Review by dkorpel — APPROVED** [review]

**Pull request merged**